x-pack/plugins/apm/server/routes/metrics.ts

```diff
@@ -18,21 +19,29 @@ const defaultErrorHandler = (err: Error) => {
 
 export function initMetricsApi(core: CoreSetup) {
   const { server } = core.http;
+
   server.route({
     method: 'GET',
     path: `/api/apm/services/{serviceName}/metrics/charts`,
     options: {
       validate: {
-        query: withDefaultValidators()
+        query: withDefaultValidators({
+          agentName: Joi.string()
+        })
       },
       tags: ['access:apm']
     },
     handler: async req => {
       const setup = setupRequest(req);
       const { serviceName } = req.params;
-      return await getAllMetricsChartData({
+      const agentName =
+        typeof req.query !== 'string' && typeof req.query.agentName === 'string'
+          ? req.query.agentName
+          : '';
```

Comment:
If `agentName` is required this can be simplified

Reply:
OK made it required everywhere but because of the types inside `req.query`, this can't be simplified much. Happy to take suggestions that pass the TS compiler though!